pkg/osversion/osversion_test.go

```diff
@@ -15,109 +13,70 @@ func TestOsversion(t *testing.T) {
 	RunSpecs(t, "Osversion Suite")
 }
 
-type testCase struct {
-	rel        string
-	maj        string
-	min        string
-	out0       string
-	out1       string
-	out2       string
-	expectsErr bool
-}
-
-func (tc *testCase) String() string {
-	str := fmt.Sprintf(
-		"%s %s.%s => [%s, %s, %s]",
-		tc.rel,
-		tc.maj,
-		tc.min,
-		tc.out0,
-		tc.out1,
-		tc.out2)
-
-	if tc.expectsErr {
-		str += " (error expected)"
-	}
-
-	return str
-}
-
 var _ = Describe("RenderOperatingSystem", func() {
-	cases := []*testCase{
-		{
-			rel: "rhcos", maj: "3", min: "0",
-			out0: "rhcos3", out1: "rhcos3.0", out2: "3.0",
-			expectsErr: false,
-		},
-		{
-			rel: "rhcos", maj: "4", min: "2",
-			out0: "rhel8", out1: "rhel8.0", out2: "8.0",
-			expectsErr: false,
-		},
-		{
-			rel: "rhcos", maj: "4", min: "4",
-			out0: "rhel8", out1: "rhel8.1", out2: "8.1",
-			expectsErr: false,
-		},
-		{
-			rel: "rhcos", maj: "4", min: "5",
-			out0: "rhel8", out1: "rhel8.2", out2: "8.2",
-			expectsErr: false,
-		},
-		{
-			rel: "rhcos", maj: "4", min: "6",
-			out0: "rhel8", out1: "rhel8.2", out2: "8.2",
-			expectsErr: false,
-		},
-		{
-			rel: "rhcos", maj: "4", min: "7",
-			out0: "rhel8", out1: "rhel8.4", out2: "8.4",
-			expectsErr: false,
-		},
-		{
-			rel: "rhcos", maj: "4", min: "8",
-			out0: "rhel8", out1: "rhel8.4", out2: "8.4",
-			expectsErr: false,
-		},
-		{
-			rel: "rhcos", maj: "4", min: "8",
-			out0: "rhel8", out1: "rhel8.4", out2: "8.4",
-			expectsErr: false,
-		},
-		{
-			rel: "rhcos", maj: "5", min: "",
-			out0: "rhcos5", out1: "rhcos5", out2: "5",
-			expectsErr: false,
-		},
-		{
-			rel: "rhcos", maj: "5", min: "1",
-			out0: "rhcos5", out1: "rhcos5.1", out2: "5.1",
-			expectsErr: false,
-		},
-	}
-
-	entries := make([]TableEntry, 0, len(cases))
-
-	for _, c := range cases {
-		entries = append(
-			entries,
-			Entry(c.String(), c),
-		)
-	}
-
 	DescribeTable("test cases",
-		func(c *testCase) {
-			out0, out1, out2, err := osversion.RenderOperatingSystem(c.rel, c.maj, c.min)
+		func(rel, maj, min, expOut0, expOut1, expOut2 string, expErr bool) {
+			out0, out1, out2, err := osversion.RenderOperatingSystem(rel, maj, min)
 
-			if c.expectsErr {
+			if expErr {
 				Expect(err).To(HaveOccurred())
 			} else {
 				Expect(err).NotTo(HaveOccurred())
-				Expect(out0).To(Equal(c.out0))
-				Expect(out1).To(Equal(c.out1))
-				Expect(out2).To(Equal(c.out2))
+				Expect(out0).To(Equal(expOut0))
+				Expect(out1).To(Equal(expOut1))
+				Expect(out2).To(Equal(expOut2))
 			}
 		},
-		entries...,
+		EntryDescription("rel=%q, maj=%q, min=%q => (%q, %q, %q) err=%t"),
+		Entry(nil,
+			"rhcos", "3", "0",
```

Comment:
I really don't like those indentions :(
Why not in a single line? it won't be longer than the `EntryDescription(...)` anyway.

Reply:
I wanted to make a separation between what the tested function receives as arguments and what it is expected to return.
If you really don't like it, we can put everything on one line.

Reply:
I think it will be cleaner in a single line.

Reply:
Everything on the same line now. PTAL.